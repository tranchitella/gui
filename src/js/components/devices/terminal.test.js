import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import TerminalDialog from './terminal';
import { undefineds } from '../../../../tests/mockData';

describe('TerminalDialog Component', () => {
  it('renders correctly', () => {
    const tree = createMount()(<TerminalDialog onCancel={jest.fn} open={true} />);
    expect(tree.html()).toMatchSnapshot();
    expect(JSON.stringify(tree.html())).toEqual(expect.not.stringMatching(undefineds));
  });
});
