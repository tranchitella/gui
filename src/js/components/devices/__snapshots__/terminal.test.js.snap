// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TerminalDialog Component renders correctly 1`] = `"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"none presentation\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthLg MuiDialog-paperFullWidth MuiPaper-elevation24 MuiPaper-rounded\\" role=\\"dialog\\"><div class=\\"MuiDialogTitle-root\\"><h2 class=\\"MuiTypography-root MuiTypography-h6\\">Terminal</h2></div><div class=\\"MuiDialogContent-root dialog-content\\" style=\\"padding: 0px;\\"><div><div dir=\\"ltr\\" class=\\"terminal xterm\\" tabindex=\\"0\\"><div class=\\"xterm-viewport\\" style=\\"background-color: rgb(0, 0, 0);\\"><div class=\\"xterm-scroll-area\\"></div></div><div class=\\"xterm-screen\\" style=\\"width: 0px; height: 0px;\\"><div class=\\"xterm-helpers\\"><textarea class=\\"xterm-helper-textarea\\" aria-label=\\"Terminal input\\" aria-multiline=\\"false\\" autocorrect=\\"off\\" autocapitalize=\\"off\\" spellcheck=\\"false\\" tabindex=\\"0\\"></textarea><span class=\\"xterm-char-measure-element\\" aria-hidden=\\"true\\" style=\\"font-family: courier-new, courier, monospace; font-size: 15px;\\">W</span><div class=\\"composition-view\\"></div></div><canvas class=\\"xterm-text-layer\\" style=\\"z-index: 0; width: 0px; height: 0px;\\" width=\\"0\\" height=\\"0\\"></canvas><canvas class=\\"xterm-selection-layer\\" style=\\"z-index: 1; width: 0px; height: 0px;\\" width=\\"0\\" height=\\"0\\"></canvas><canvas class=\\"xterm-link-layer\\" style=\\"z-index: 2; width: 0px; height: 0px;\\" width=\\"0\\" height=\\"0\\"></canvas><canvas class=\\"xterm-cursor-layer\\" style=\\"z-index: 3; width: 0px; height: 0px;\\" width=\\"0\\" height=\\"0\\"></canvas></div></div></div></div><div class=\\"MuiDialogActions-root MuiDialogActions-spacing\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Close</span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;
